ethicalgardeners/metricscollector.py

```diff
@@ -0,0 +1,320 @@
+                        self.metrics["num_cells_pollution_above_75"] += 1
+                        self.metrics["num_cells_pollution_above_50"] += 1
+                        self.metrics["num_cells_pollution_above_25"] += 1
+                    elif (cell.pollution*100/max_pollution) > 75:
+                        self.metrics["num_cells_pollution_above_75"] += 1
+                        self.metrics["num_cells_pollution_above_50"] += 1
+                        self.metrics["num_cells_pollution_above_25"] += 1
+                    elif (cell.pollution*100/max_pollution) > 50:
+                        self.metrics["num_cells_pollution_above_50"] += 1
+                        self.metrics["num_cells_pollution_above_25"] += 1
+                    elif (cell.pollution*100/max_pollution) > 25:
+                        self.metrics["num_cells_pollution_above_25"] += 1
+
+        if len(grid_world.grid) > 0:
+            self.metrics["avg_pollution_percent"] /= num_cells
+
+        self.metrics["rewards"] = rewards
+        for agent, reward in rewards.items():
+            if agent in self.metrics["accumulated_rewards"]:
+                self.metrics["accumulated_rewards"][agent] += reward
+            else:
+                self.metrics["accumulated_rewards"][agent] = reward
+        self.metrics["agent_selection"] = agent_selection
+
+    def reset_metrics(self):
+        """
+        Reset all metrics to their initial values.
+
+        This method should be called when the environment is reset to ensure
+        metrics are tracked correctly for each new simulation run.
+        """
+        self.metrics = {
+            "step": 0,
+            "total_planted_flowers": 0,
+            "num_planted_flowers_per_agent": {},
+            "total_harvested_flowers": 0,
+            "num_harvested_flowers_per_agent": {},
+            "avg_pollution_percent": 0.0,
+            "num_cells_pollution_above_90": 0,
+            "num_cells_pollution_above_75": 0,
+            "num_cells_pollution_above_50": 0,
+            "num_cells_pollution_above_25": 0,
+            "rewards": {},
+            "accumulated_rewards": {},
+            "agent_selection": None,
+        }
+
+    def export_metrics(self):
+        """
+        Export collected metrics to a local file.
+
+        This method exports the current metrics to a CSV file in the specified
+        output directory if export_on is True. A new file is created for each
+        run of the program, and metrics are appended to this file at each call.
+        """
+        if self.export_on:
+            try:
+                import csv
+            except ImportError:
+                raise ImportError(
+                    "Error while importing csv module. "
+                )
+            try:
+                import os
+            except ImportError:
+                raise ImportError(
+                    "Error while importing os module. "
+                )
```

Comment:
Same as for `time`, do not worry about ImportErrors here. Otherwise, we would need them for any single `import` line we do...

ImportError is particularly useful when you want to catch the error, for example to provide a fallback mechanism (as you do for `agent_selector`, typically).